### PR TITLE
Double link editors

### PR DESCRIPTION
**Pull request by RabsRincon**
This PR aims to ease the finding of the lines of code that generate certain assembly / vice versa
It comes in two flavours. 
As a context menu option and as an hover event.

- The first has a shortcut of Ctrl + F10 (I'm more than open to change it, I only used it as a placeholder)

- The second shows the line releated to the one that the cursor ir currently hover, and it's controlled by a new checkbox on the settings page

### CODE REVIEW

File: static/settings.js [hunk 1]
```diff
@@ -78,13 +78,7 @@ define(function (require) {
...
     function setupSettings(root, settings, onChange) {
         settings = settings || {};
-        if (settings.delayAfterChange === undefined)
-            settings.delayAfterChange = 750;
-        if (settings.colouriseAsm === undefined)
-            settings.colouriseAsm = true;
-        if (settings.hoverShowSource === undefined)
-            settings.hoverShowSource = true;
-
+
```
Comment (RabsRincon) [inline static/settings.js]: Were those lines not necessary or am I missing their replacement somewhere else on the code? I got a merge conflict so I guess they were not necessary

**Comment by mattgodbolt** [comment]
(Will review more later) - but those lines are now unnecessary and I
removed as part of some clean up. Thanks for checking.

**Comment by RabsRincon** [comment]
I will switch from selecting the desired line to
https://microsoft.github.io/monaco-editor/playground.html#interacting-with-the-editor-rendering-glyphs-in-the-margin
as I think it fits better

**Comment by mattgodbolt** [comment]
Oh cool! Thanks...sorry I didn't get a chacne to review; will take a look
when you're happy with the glyph version!

**Comment by RabsRincon** [comment]
This feature is now ready for review!
@mattgodbolt let me know if I should change anything or add/remove some feature.

File: static/compiler.js [hunk 2]
```diff
@@ -96,7 +99,30 @@ define(function (require) {
...
+            contextMenuGroupId: 'navigation',
+            contextMenuOrder: 1.5,
+            run: function(ed) {
+                var desiredLine = ed.getPosition().lineNumber - 1;
+                self.eventHub.emit('editorSelectLine', self.sourceEditorId, self.assembly[desiredLine].source);
+            }
+        });
+
+        this.outputEditor.onMouseMove(function (e) {
+            setTimeout(function() {
```
Comment (RabsRincon) [inline static/compiler.js]: @mattgodbolt What do you think about this timeout as to throttle the MouseMove event and not comput it every "tick"? There is also one on the other text editor

**Comment by mattgodbolt** [comment]
Cool; can you first rebase against the latest master?  NB I've deleted index.html in my branch: I think it should be straightforward to port any changes to the `index.pug` file. Let me know if you hit any issues!

**Comment by mattgodbolt** [comment]
Re: setTimeout - can you use instead _.throttle() or similar? Otherwise
you're not throttling anything, you're just setting many many timeouts!

**Comment by RabsRincon** [comment]
I think both things are resolved now (I ran into some problems with the rebase, let me know if everything is ok please)

**Comment by RabsRincon** [comment]
Ok @mattgodbolt sorry for all the commits suddenly. Everything should be good for a review now

**Comment by mattgodbolt** [comment]
Hi Ruben; sorry I didn't get to take a look last night. Looking now :)

**Comment by mattgodbolt** [comment]
Wow; it looks so cool! Thanks so much. Will spend a bit of time looking at the code, but this is nearly good to go. There are some issues with multiple compilers open, only one of them updates (e.g. go to a URL like http://localhost:10240/#g:!((g:!((g:!((h:codeEditor,i:(j:1,source:'int+square(int+num)+%7B%0A++return+num+*+num%3B%0A%7D'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((g:!((h:compiler,i:(compiler:/home/mgodbolt/.fighome/runtime/gcc/6.3.0-1/bin/g%2B%2B,filters:(b:'0',commentOnly:'0',directives:'0'),options:'',source:1),l:'5',n:'0',o:'%231+with+/home/mgodbolt/.fighome/runtime/gcc/6.3.0-1/bin/g%2B%2B',t:'0')),k:50,l:'4',m:50,n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:/home/mgodbolt/.fighome/runtime/gcc/6.3.0-1/bin/g%2B%2B,filters:(b:'0',commentOnly:'0',directives:'0',intel:'0'),options:'-O2',source:1),l:'5',n:'0',o:'%231+with+/home/mgodbolt/.fighome/runtime/gcc/6.3.0-1/bin/g%2B%2B',t:'0')),l:'4',m:50,n:'0',o:'',s:0,t:'0')),k:50,l:'3',n:'0',o:'',t:'0')),l:'2',n:'0',o:'',t:'0')),version:4 ) and mouse over the source code, only one of the two compilers updates. I can take a look at this though!

Thanks so much!

**Comment by mattgodbolt** [comment]
Another minor bug: disabling the "Colourise" option in settings, then re-enabling it and we lose the decorations until you reload the page. I wonder if the two sets of decorations are fighting? I notice the decoration is marked as "isWholeLine" (line 513 of compiler.js) but I only see a little blob to the left of the code? Is it meant to highlight the whole line?

**Comment by mattgodbolt** [comment]
Oh it seems the "hover" setting is being lost when I open and close the settings box :) I bet my recent changes broke it, sorry!

**Comment by mattgodbolt** [comment]
(the setting was a quick fix: you'd misspelt the name in settings.js:
```
        add(root.find('.hoverShowSource'), 'hoverShowSource', true, Checkbox);
```

fixes it :)

**Comment by mattgodbolt** [comment]
This is great; I'm going to merge and we can fix up any minor issues in master! Thank you so much!

**Comment by mattgodbolt** [comment]
I'd like to credit you in the README, what's the best way to refer to you? :) As @RabsRincon ?

**Comment by mattgodbolt** [comment]
Oh cool; please do PR against master - that would be awesome :) I'll use Rubén and your @RabsRincon in the credits! Hooray :)

**Pull request merged**